### ISSUE TITLE
Rapid tag and update games

### ISSUE THREAD

**Issue by gajop** [Spring-Chobby/Chobby#56]
Tagging games should download and install them and also save this information so that they can be updated.
VFS.TagGame()
VFS.UntagGame()
VFS.UpdateGames()
VFS.ListTaggedGames()

**Comment by ashdnazg** [comment]
I think tagging can easily be implemented in the lua side and shouldn't be in the rapid lib itself.

**Comment by gajop** [comment]
You *can* do it in Lua, but there's a benefit if tagged (installed) games are a shared concept in a predefined format that other lobbies/games can rely on.
Same with validAIs and a number of other things

**Comment by ashdnazg** [comment]
Makes sense to store the tags in a shared place, but I wouldn't put it into rapid

**Comment by gajop** [comment]
Maybe I misunderstand you, but I feel it should be part of the pr-downloader library, like it used to be with "rapid pin" and "rapid unpin" https://github.com/tvo/rapid#usage

**Comment by ashdnazg** [comment]
That's exactly what I don't want to put in the pr-downloader library.
I want it to just deal with downloading and rapid, since this can't be done in lua.
Lua code is easier to change, fix, test, code, everything basically.

So if you can tag/untag things in lua, that's far better than implementing it inside pr-downloader.

**Comment by ashdnazg** [comment]
P.S. it's not a coincidence that the tool written in python developed this functionality before the main C++ code

**Comment by ashdnazg** [comment]
btw, it can make sense to put this lua code in the base content to enforce the predefined format.

**Comment by gajop** [comment]
If we did it in Lua, that would be a reasonable thing to do. Still, I think it'd be unfortunate to limit this only to inengine stuff. 

The tool itself as well as any other lobby using pr-downloader would have to implement it from scratch and that leaves room for misinterpretation, mistakes and simply not doing it (like no one has done it since).

**Comment by ashdnazg** [comment]
a valid concern, I'll think about it

**Comment by gajop** [comment]
Well https://github.com/spring/pr-downloader/issues/12 exists, so let's invoke @abma to see what he thinks about it.

**Comment by ashdnazg** [comment]
perhaps unitsync is a better place for this, what with reading info etc.

**Comment by abma** [comment]
imo this can be already implemented without changing c++ code:

- for upgrade just download a tag again
- store the tags in a config entry

**Comment by gajop** [comment]
I feel you didn't read the entire issue. Yes it can be implemented easily in Lua, but question is if that's desirable. The arguments for and against are above.

**Comment by abma** [comment]
> Yes it can be implemented easily in Lua, but question is if that's desirable. The arguments for and against are above.

yes, thats imo desireable. storing / changing a string is very few code and can be easily implemented in lua or by lobbies itself when needed. creating / changing an api is much more hassle and the lua api in spring is already heavily bloated.
